pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -90,6 +91,14 @@ def __enforce_datamodule_dataloader_override(self, train_dataloader, val_dataloa
                 'You cannot pass train_dataloader or val_dataloaders to trainer.fit if you supply a datamodule'
             )
 
+    def _validate_data_hooks(self, model):
+        # Raise Misconfiguration exception since these hooks are not supported in DP mode
+        # TODO: Remove this blocker once batch transfer to device is integrated in Lightning for DP mode.
+        batch_transfer_hooks = ['on_before_batch_transfer', 'transfer_batch_to_device', 'on_after_batch_transfer']
+        for hook in batch_transfer_hooks:
+            if self.trainer.accelerator_connector.use_dp and is_overridden(hook, model):
```

Comment:
I think this works?

```suggestion
            if self.trainer.use_dp and is_overridden(hook, model):
```

Reply:
deprecated: https://github.com/PyTorchLightning/pytorch-lightning/blob/ae6ce17598cde260f980079b5b3a4274c19eb7a1/pytorch_lightning/trainer/deprecated_api.py#L73-L75